tests/undefined_variable_detection.rs

```diff
@@ -10,7 +10,7 @@ use std::ffi::OsStr;
 fn parse_and_codegen(src: &'static str) -> Namespace {
     let mut cache = FileResolver::new();
     cache.set_file_contents("test.sol", src.to_string());
-    let mut ns = parse_and_resolve(OsStr::new("test.sol"), &mut cache, Target::Ewasm);
+    let mut ns = parse_and_resolve(OsStr::new("test.sol"), &mut cache, Target::EVM);
```

Comment:
I think undefined variable detection depends on codegen. Will this test work even if we use `Target::EVM`?

Reply:
Same as above. We're keeping the code `src/codegen/...`.